spec/support/ssl_helper.rb

```diff
@@ -12,16 +12,41 @@ def initialize
     def next
       @iter.next
     rescue StopIteration
-      @keystash << OpenSSL::PKey::RSA.new(2048)
+      @keystash << generate_key
       save
       @keystash.last
     end
 
+    def generate_key
+      case (rand * 4).to_i
+      when 0
+        OpenSSL::PKey::RSA.new(2048)
+      when 1
+        generate_ecdsa_key('prime256v1')
+      when 2
+        generate_ecdsa_key('secp384r1')
+      when 3
+        generate_ecdsa_key('secp521r1')
+      end
+    end
+
+    def generate_ecdsa_key(curve)
+      k = OpenSSL::PKey::EC.new(curve)
+      k.generate_key
+      Acme::Client::CertificateRequest::ECKeyPatch.new(k)
```

Comment:
Its more of an internal object, so you can use `OpenSSL::PKey::EC` normally here.

`ECKeyPatch` is only there to avoid monkey patching openssl. Its because `OpenSSL::PKey::EC` does not respect the interface of `OpenSSL::PKey::RSA` and it break compatibility with the `OpenSSL::X509::Request`.